### PR TITLE
remove double-wrapped safe method calling from ThingManager

### LINKED ISSUE #2986: Handlers taking more than 5000ms to run
There are a growing number of problems showing up on the OH2 forum about handlers taking more than 5000ms - see -:

https://community.openhab.org/t/handler-takes-more-than-5000ms-for-processing-event/22158/15

This seems to affect a wide range of bindings so I would guess it could be framework related.

### PR DESCRIPTION
**Pull request by sjka**
...when initializing Things.

fixes #2986
Signed-off-by: Simon Kaufmann <simon.kfm@googlemail.com>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java [hunk 1]
```diff
@@ -977,26 +967,8 @@ protected void addThingHandlerFactory(ThingHandlerFactory thingHandlerFactory) {
...
-                        registerHandler(thing, thingHandlerFactory);
-                        initializeHandler(thing);
-                        return null;
-                    }
-                });
-            } catch (TimeoutException e) {
-                logger.warn("Registering a handler for thing '{}' takes more than {}ms.", thing.getUID(),
-                        SafeMethodCaller.DEFAULT_TIMEOUT);
-            } catch (Exception ex) {
-                ThingStatusInfo statusInfo = buildStatusInfo(ThingStatus.UNINITIALIZED,
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java]: Hm, this method is also called by thingUpdated() and thingAdded() and both these methods do NOT catch any runtime exceptions - so the exception handling that you remove here isn't done anywhere else. Am I missing something?

Comment (sjka) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java]: both `registerHandler(...)` and `initializeHandler(...)` do exception handling themselves, also setting the thing status accordingly. What are you missing here?

Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java]: Ok, then I missed looking down the code, I only looked up. I somehow expected that this code here was of any use before, but apparently it wasn't...

Just a question which I asked myself (but which is not directly related to this PR): What happens in the (unlikely, but possible) case that handler.initialize() throws an exception AFTER MORE than 5 seconds?

**Comment by sjka** [comment]
> What happens in the (unlikely, but possible) case that handler.initialize() throws an exception AFTER MORE than 5 seconds?

Short answer: The (angry) user will reboot 😉  

Long answer: The thing is going to stay in INITIALIZING state. In the logs there will be a timeout entry and another hopefully meaningful error. We could (in a separate PR) think about setting the thing's state back to UNITIALIZED (HANDLER_INITIALIZING_ERROR: it took too long). But we got to be careful about preventing race conditions which could accidentally lead to INITIALIZING -> ONLINE -> UNINITIALIZED.

**Pull request merged**